lib/commands/app-management.js

```diff
@@ -163,6 +162,26 @@ commands.installApp = async function (appPath, options = {}) {
   }
 };
 
+/**
+*  Check the currently installed version of the application
+*   @param {string} appId
+*   @throws {Error} if device is unreachable
+*   @returns {string[]} the versions of applications installed under that package
+*/
+commands.findVersions = async function (appId) {
```

Comment:
It doesn't implement any part of the w3c webdriver specification; just something that I've run into where I and my team need to have version information exposed and wanted to avoid having yet another service running. It didn't seem like there was a way to expose such information with appium currently.  

If this is wrong, or we should just tackle the problem from adding another service, let me know, I can close the PR and go about it that way.

Also, thank you for pointing out the getApkInfo method! 

Cheers,

Ben Straub

Reply:
There is a possibility to execute any shell commands in general: https://github.com/appium/appium/blob/e4e59193f44a469f719d766442296b9eafc97de0/docs/en/writing-running-appium/android/android-shell.md

In case you need something special then you could consider adding this functionality under `mobile:` helper. Check 
https://github.com/appium/appium-uiautomator2-driver/blob/661dcd6df615e267fb94bb0d2fdacc11201b75b5/lib/commands/general.js#L64

and 

https://github.com/appium/appium-android-driver/blob/263afd668a4e2e4e8b404f4d5f05f570b18b4a1d/lib/commands/execute.js#L15

Reply:
This is excellent! My only concern is if there is parity for ios; I wrote a little addition to the appium-xcuitest that uses "ideviceinstaller -l" to scrape version info. . .  is there a way currently exposed in appium to retrieve application info for ios?

Reply:
In any case, it seems like you've proven this addition isn't necessary - I will close this PR.